src/resource/resource.test.ts

```diff
@@ -502,6 +510,52 @@ describe("getResourceInfo", () => {
       { fetch: mockFetch }
     );
 
+    expect(solidDatasetInfo.internal_resourceInfo.linkedResources).toEqual({
+      acl: "https://some.pod/aclresource.acl",
+      // Unfortunately the library we uses lowercases `rel`s:
+      // https://github.com/jhermsmeier/node-http-link-header/issues/22
+      // At the time of writing, only our own library inspects these, so it's something only we
+      // need to keep in mind:
+      // "http://www.w3.org/ns/solid/terms#podOwner": "https://some.pod/profile#WebId",
+      "http://www.w3.org/ns/solid/terms#podowner":
+        "https://some.pod/profile#WebId",
+    });
+  });
+
+  it("exposes when no Resources were linked", async () => {
+    const mockFetch = jest.fn(window.fetch).mockResolvedValue(
+      mockResponse(undefined, {
+        url: "https://arbitrary.pod",
+      })
+    );
+
+    const solidDatasetInfo = await getResourceInfo(
+      "https://some.pod/container/resource",
+      { fetch: mockFetch }
+    );
+
+    expect(
+      solidDatasetInfo.internal_resourceInfo.linkedResources
+    ).toBeUndefined();
```

Comment:
Would it make sense that if there are no links defined, the `linkedResources` parameter is an empty map, as opposed to `undefined` ? What would be the most convenient to test for ?

Reply:
Yes, good that you bring this up, because that's something we should discuss. I initially went this route, but there was one area where this is problematic: Resources newly created in a Container.

What we currently do is, after creating a Resource in a Container, we manually initialise a new dataset, include the data we just saved in it, and set the ResourceInfo on it. However, this is not possible for `linkedResources` - we can only infer those by actually making a (HEAD) request to the newly-created Resources and inspecting its headers. But then we run into the problem that the current user might not have Read access to the newly-created Resource, e.g. if it was created in a Container like the inbox, to which they only have Append access.

Now that I write this, I think maybe the best solution would be to make that request anyway, and return `null` from the functions that create something in a Container if the current user does not have Read access to it. After all, there's not much you can do with the returned value anyway, and if you want to inspect what data you _sent_ to the Container, then you can just inspect the dataset you passed as input.

What do you think?

Reply:
Just to be sure that I understand correctly, your proposal in this case would be to: 
1. Create the resource
2. Make the `HEAD` request to get the data for the `resourceInfo`
3. If the HEAD request fails (i.e. the user doesn't have read access), `null` is returned. Otherwise, the parsing of the `resourceInfo` happens as usual.
4. The `linkedResources` property is always defined for a non-`null` `resourceInfo`, being `{}` if no links are defined, and a `Record<string, string[]>` otherwise (or the type that you see appropriate).

Is that correct ?

Reply:
Yes, that is correct. (And technically, `{}` is a `Record<string, string[]>` as well, just an empty one :stuck_out_tongue: ) Steps 1-3 would be in `saveInContainer()` and similar functions. `saveSolidDatasetAt()` and the likes (that do not create things in Containers) already get the required data from the Response.

Reply:
- Returning `null` for Resources created in a Container without Read access: 8e80579af1e2ee67af4fa4d8b6f39983e08a863b
- Don't allow `undefined` for `linkedResources`: 9fa30d11e48e1eb9f30686dba3996bf9e9413ea4

Reply:
Ah, forgot running the build and missed a couple of things:

- Returning `null` for `createFileInContainer` as well: bb511bd89e30306cd42f0f5614289038ab92c196 and f9fd294824ac51acd4e828ee6b974b1d7f156782
- Some places that stilled had `undefined` for `linkedResources`: 2186b8ae1f57b6da575bbaf95c61d13bbda4ff5f and fcc8faf466a579c3b4148fa10ba1d769cc32da62